tests/test_datatypes.py

```diff
@@ -41,6 +41,12 @@ def test_number():
     assert t.formatted(v) == '3'
     with pytest.raises(ValueError):
         t.validate(12)
+    
+    t = Datatype.fromvalue({'base': 'nonNegativeInteger'})
+    v = t.parse('-3')
+    with pytest.raises(ValueError):
```

Comment:
The `ValueError` is already raised when `parse` is called.

Reply:
Yes, just realised. So you want to separate parsing from validating then?

Reply:
No, only the test should be adapted by moving the `pytest.raises` context around the `parse` call. I think `parse` triggering validation implitly is totally legit semantics.

Reply:
Other datatypes do that, too: https://github.com/cldf/csvw/blob/f9434800b63c1067d3409e743273c19bfbf52fdd/tests/test_datatypes.py#L133-L135